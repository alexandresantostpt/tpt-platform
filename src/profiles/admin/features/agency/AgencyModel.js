import { config } from '@config'

import { appThemes } from './appThemes'

const DEFAULT_THEME = 2

class AgencyModel {
    constructor({ __v, _id, appTheme, businessName, cnpj, image, phoneContact, phoneEmergency, socialName }) {
        this.__v = __v
        this._id = _id
        this._appTheme = appTheme || appThemes[DEFAULT_THEME]
        this._businessName = businessName
        this._cnpj = cnpj
        this._image = image
        this._phoneContact = phoneContact
        this._phoneEmergency = phoneEmergency
        this._socialName = socialName
    }

    get v() {
        return this.__v
    }

    get id() {
        return this._id
    }

    get appTheme() {
        return this._appTheme
    }

    get businessName() {
        return this._businessName
    }

    get cnpj() {
        return this._cnpj
    }

    get image() {
        return this._image
    }

    get phoneContact() {
        return this._phoneContact
    }

    get phoneEmergency() {
        return this._phoneEmergency
    }

    get socialName() {
        return this._socialName
    }

    getImage() {
        return this.image && `${config.api.url}/agency/download/${this.image}`
    }

    toDTO() {
        return {
            __v: this.v,
            _id: this.id,
            appTheme: this.appTheme || null,
            businessName: this.businessName || null,
            cnpj: this.cnpj || null,
            image: this.image || null,
            phoneContact: this.phoneContact || null,
            phoneEmergency: this.phoneEmergency || null,
            socialName: this.socialName || null
        }
    }

    toForm() {
        return {
            __v: this.v,
            _id: this.id,
            appTheme: this.appTheme,
            businessName: this.businessName,
            cnpj: this.cnpj,
            image: this.image,
            phoneContact: this.phoneContact,
            phoneEmergency: this.phoneEmergency,
            socialName: this.socialName
        }
    }
}

export default AgencyModel
