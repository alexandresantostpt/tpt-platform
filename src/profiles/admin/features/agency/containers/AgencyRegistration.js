import { connect } from 'react-redux'

import { requestGetSingleAction, requestSaveDataAction } from '../actions'

import { getObjSelector } from '../selectors'

import AgencyRegistration from '../pages/AgencyRegistration'

const mapStateToProps = state => ({
    agency: getObjSelector(state)
})

const mapDispatchToProps = { requestGetSingleAction, requestSaveDataAction }

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(AgencyRegistration)
