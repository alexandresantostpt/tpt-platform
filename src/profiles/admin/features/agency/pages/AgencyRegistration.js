import React, { useRef, useEffect, useState } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import App from '@containers/App'
import Container from '@containers/Container'
import Summary from '@containers/Summary'

import Avatar from '@/components/avatar/Avatar'
import FormBuilder from '@/components/form/FormBuilder'
import FormButtonContainer from '@/components/form/container/button/FormButtonContainer'
import Button from '@/components/button/Button'

import { colors } from '@helpers/colors'
import { getUser } from '@/utils/auth'

import { i18n } from '@i18n'

import AgencyModel from '../AgencyModel'
import { agencyCompleteFormFields } from '../fields'

import { routesRedefinePassword } from '@features/login/routes'
import { routes as routesTravel } from '@features/travel/routes'

import profileAvatar from '@img/avatars/persona1.png'
import UploadArea from '../components/UploadArea'
import ColorPicker from '../components/color-picker/ColorPicker'

import { appThemes } from '../appThemes'

const HeaderContainer = styled.div`
    align-items: center;
    display: flex;
    flex-direction: column;
`

const BodyContainer = styled.div`
    padding: 2.25rem;
`

const Title = styled.div`
    color: ${colors.text};
    font-size: 2.8125rem;
    font-weight: 300;
    margin: 2.8125rem 0;
`

const Text = styled(Title)`
    font-size: 1.5625rem;
`

const agencyInitialValues = agency => (agency ? agency.toForm() : {})

const AgencyRegistration = ({ agency, requestGetSingleAction, requestSaveDataAction }) => {
    const user = getUser()
    const form = useRef()
    const [image, updateImage] = useState(null)
    const [color, updateColor] = useState(null)

    useEffect(() => {
        const id = user.agency
        requestGetSingleAction({ id })
    }, [])

    const redirectTo = user && user.active ? routesTravel[0].path : routesRedefinePassword.path

    const callSaveData = values => {
        const agencyModelData = color ? { ...values, appTheme: color } : { ...values }
        requestSaveDataAction({ data: new AgencyModel(agencyModelData).toDTO(), image, redirectTo: redirectTo })
    }

    const onSaveClick = () => form.current.form.submit()

    const getImage = () => agency && agency.getImage()

    return (
        <App>
            <Container>
                <HeaderContainer>
                    <Avatar alt={user && user.name} image={(user && user.getImage()) || profileAvatar} size={10} />
                    <Title>{i18n.t('titles.greetings', { name: user && user.name })}</Title>
                </HeaderContainer>
                <BodyContainer>
                    <Text>{i18n.t('info.fillFields')}</Text>
                    <FormBuilder
                        fields={agencyCompleteFormFields}
                        fieldsToValidate={[...agencyCompleteFormFields]}
                        initialValues={agencyInitialValues(agency)}
                        onSubmit={callSaveData}
                        ref={process.env.__TEST__ ? null : form}
                    />
                    <UploadArea addImage={updateImage} image={getImage()} />
                    <ColorPicker
                        color={color || (agency && agency.appTheme)}
                        colorList={appThemes}
                        placeholder={i18n.t('placeholders.colors')}
                        updateColor={updateColor}
                    />
                    <FormButtonContainer>
                        <Button color={colors.purple} disabled={false} onClick={onSaveClick} text={i18n.t('buttons.save')} />
                    </FormButtonContainer>
                </BodyContainer>
            </Container>
            <Summary />
        </App>
    )
}

AgencyRegistration.defaultProps = {
    agency: null
}

AgencyRegistration.propTypes = {
    agency: PropTypes.objectOf(AgencyModel),
    requestGetSingleAction: PropTypes.func.isRequired,
    requestSaveDataAction: PropTypes.func.isRequired
}

export default AgencyRegistration
