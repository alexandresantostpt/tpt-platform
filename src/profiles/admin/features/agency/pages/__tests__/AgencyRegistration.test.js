import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'

import AgencyRegistration from '../AgencyRegistration'

import AgencyModel from '../../AgencyModel'

describe('Tests for AgencyRegistration component', () => {
    const agency = new AgencyModel({
        __v: 0,
        _id: '5c52ebcd5fc7c9b1433ab704',
        agency: {},
        appTheme: '#FFFFFF',
        businessName: 'This is my test',
        cnpj: '12345678909876',
        createdAt: '2019-03-01T12:35:34.965Z',
        logoTipo: 'logoTipo.png',
        phoneContact: '191234566',
        phoneEmergency: '1932465789',
        socialName: 'Test',
        updatedAt: '2019-03-01T12:35:34.965Z',
        user: {}
    })

    it('Should render withou crash', () => {
        const renderer = new ShallowRenderer()
        const tree = renderer.render(
            <AgencyRegistration agency={agency} requestGetSingleAction={jest.fn()} requestSaveDataAction={jest.fn()} />
        )
        expect(tree).toMatchSnapshot()
    })
})
