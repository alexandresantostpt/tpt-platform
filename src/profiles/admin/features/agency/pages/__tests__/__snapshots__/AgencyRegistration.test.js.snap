// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for AgencyRegistration component Should render withou crash 1`] = `
<App
  travelScript={null}
>
  <Container>
    <ForwardRef>
      <Avatar
        alt={null}
        image="persona1.png"
        onClick={null}
        size={10}
      />
      <ForwardRef>
        Olá, !
      </ForwardRef>
    </ForwardRef>
    <ForwardRef>
      <ForwardRef>
        Para continuar, preencha os campos abaixo:
      </ForwardRef>
      <ForwardRef
        disabled={false}
        fieldChildren={null}
        fields={
          Array [
            Object {
              "component": "input",
              "id": "socialName",
              "label": "socialName",
              "maxLength": 120,
              "name": "socialName",
              "required": true,
              "type": "text",
            },
            Object {
              "component": "input",
              "id": "businessName",
              "label": "businessName",
              "maxLength": 120,
              "name": "businessName",
              "required": true,
              "type": "text",
            },
            Object {
              "component": "input",
              "id": "cnpj",
              "label": "cnpj",
              "mask": Object {
                "blocks": Array [
                  2,
                  3,
                  3,
                  4,
                  2,
                ],
                "delimiters": Array [
                  ".",
                  ".",
                  "/",
                  "-",
                ],
              },
              "maxLength": 18,
              "name": "cnpj",
              "required": true,
              "type": "mask",
              "validation": "cnpj",
            },
            Object {
              "component": "input",
              "id": "phoneEmergency",
              "label": "phoneEmergency",
              "maxLength": 30,
              "name": "phoneEmergency",
              "required": true,
              "type": "text",
            },
            Object {
              "component": "input",
              "id": "phoneContact",
              "label": "phoneContact",
              "maxLength": 30,
              "name": "phoneContact",
              "required": true,
              "type": "text",
            },
          ]
        }
        fieldsToValidate={
          Array [
            Object {
              "component": "input",
              "id": "socialName",
              "label": "socialName",
              "maxLength": 120,
              "name": "socialName",
              "required": true,
              "type": "text",
            },
            Object {
              "component": "input",
              "id": "businessName",
              "label": "businessName",
              "maxLength": 120,
              "name": "businessName",
              "required": true,
              "type": "text",
            },
            Object {
              "component": "input",
              "id": "cnpj",
              "label": "cnpj",
              "mask": Object {
                "blocks": Array [
                  2,
                  3,
                  3,
                  4,
                  2,
                ],
                "delimiters": Array [
                  ".",
                  ".",
                  "/",
                  "-",
                ],
              },
              "maxLength": 18,
              "name": "cnpj",
              "required": true,
              "type": "mask",
              "validation": "cnpj",
            },
            Object {
              "component": "input",
              "id": "phoneEmergency",
              "label": "phoneEmergency",
              "maxLength": 30,
              "name": "phoneEmergency",
              "required": true,
              "type": "text",
            },
            Object {
              "component": "input",
              "id": "phoneContact",
              "label": "phoneContact",
              "maxLength": 30,
              "name": "phoneContact",
              "required": true,
              "type": "text",
            },
          ]
        }
        initialValues={
          Object {
            "__v": 0,
            "_id": "5c52ebcd5fc7c9b1433ab704",
            "appTheme": "#FFFFFF",
            "businessName": "This is my test",
            "cnpj": "12345678909876",
            "image": undefined,
            "phoneContact": "191234566",
            "phoneEmergency": "1932465789",
            "socialName": "Test",
          }
        }
        onSubmit={[Function]}
        rangeDates={Object {}}
        suggestions={Array []}
        type={null}
      />
      <UploadArea
        addImage={[Function]}
        image=""
      />
      <OnClickOutside(ColorPicker)
        color="#FFFFFF"
        colorList={
          Array [
            Object {
              "primary": "#ed6ea0",
              "primaryDark": "#ed6ea0",
              "primaryLight": "#ec8c69",
              "text": "#3d3e46",
            },
            Object {
              "primary": "#B6CEE8",
              "primaryDark": "#B6CEE8",
              "primaryLight": "#F578DC",
              "text": "#3d3e46",
            },
            Object {
              "primary": "#B19FFF",
              "primaryDark": "#5271C4",
              "primaryLight": "#ECA1FE",
              "text": "#3d3e46",
            },
            Object {
              "primary": "#00dbde",
              "primaryDark": "#00dbde",
              "primaryLight": "#fc00ff",
              "text": "#3d3e46",
            },
            Object {
              "primary": "#48c6ef",
              "primaryDark": "#48c6ef",
              "primaryLight": "#6f86d6",
              "text": "#3d3e46",
            },
            Object {
              "primary": "#37ecba",
              "primaryDark": "#37ecba",
              "primaryLight": "#72afd3",
              "text": "#3d3e46",
            },
            Object {
              "primary": "#e0e0e0",
              "primaryDark": "#e0e0e0",
              "primaryLight": "#bcbcbc",
              "text": "#3d3e46",
            },
            Object {
              "primary": "#a8caba",
              "primaryDark": "#a8caba",
              "primaryLight": "#5d4157",
              "text": "#3d3e46",
            },
            Object {
              "primary": "#29323c",
              "primaryDark": "#29323c",
              "primaryLight": "#485563",
              "text": "#3d3e46",
            },
          ]
        }
        eventTypes={
          Array [
            "mousedown",
            "touchstart",
          ]
        }
        excludeScrollbar={false}
        outsideClickIgnoreClass="ignore-react-onclickoutside"
        placeholder="Escolha um padrão de cores para seu aplicativo..."
        preventDefault={false}
        stopPropagation={false}
        updateColor={[Function]}
      />
      <ForwardRef>
        <Button
          color="#8187e1"
          disabled={false}
          fill={false}
          onClick={[Function]}
          text="Salvar"
        />
      </ForwardRef>
    </ForwardRef>
  </Container>
  <Summary />
</App>
`;
